### DISCUSSION TITLE
Encrypt Buffer/UInt8Array with public keys?

### DISCUSSION THREAD

**Discussion by eep-oop-urp** [openpgpjs/openpgpjs#1537] [General]
I've been struggling with how to use this library in an AWS lambda using NodeJS 14.x and TypeScript.

I attempted to write code that downloads a PDF from S3 using the aws-sdk and encrypt it with a public key, but to no avail.

In the documentation, there are examples of encrypting a UInt8Array with a password and of encrypting a WebStream with public keys. I can't use the latter because ReadableStream is not available in NodeJS 14 and I can't upgrade to NodeJS 18.

I didn't find an example of encrypting a UInt8Array with a public key. Is this possible? If so, is there an example?

**Comment by twiss** [comment]
Yes, it's possible. Please see the [documentation](https://docs.openpgpjs.org/global.html#encrypt), or you can combine aspects from different examples.

**Comment by larabr** [comment]
Hi @eep-oop-urp , yes it is possible, the examples only show the most popular combination of options, but you can "mix and match" them.
To encrypt binary data with a public key, just pass `encryptionKeys` instead of `passwords`, namely:
```js
    const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });
    const data = new Uint8Array(..);

    const encrypted = await openpgp.encrypt({
        message: await openpgp.createMessage({ binary: data }),
        encryptionKeys: publicKey
    });
```
You can also pass `signingKeys` if you need to sign the message.

**Reply by eep-oop-urp** [reply]
Thanks, I ended up with the following:

```javascript
  const bodyMessage = await createMessage({ binary: (pdfBody.Body as Uint8Array) });
  return encrypt({
    message: bodyMessage,
    encryptionKeys: publicKey,
    format: 'binary',
  });
```
